aten/src/ATen/AccumulateType.h

```diff
@@ -51,147 +52,231 @@
 
 namespace at {
 
-template <typename T, bool is_cuda>
-struct AccumulateType {};
+template <typename T, c10::DeviceType D>
+struct AccumulateTypeDevice {};
+
+template <>
```

Comment:
nit: I would use macro to make these definitions more compact and easier to read with something like:
```
MPS_ACC_TYPE(BFloat16, float);
MPS_ACC_TYPE(Half, float);
...
MPS_ACC_TYPE(int8_t, int64_t);
...
```

Reply:
Love this suggestion! Done.